third-culture-software/bhima
# Update asset inventory scans

More updates to the assets scanning
- Adds counts for items scanned vs total number of items to scan + percentage to Needed Asset Scans campaign reports.
- Fixed issue with needed asset scans reports related to not showing correct latest scans

Closes https://github.com/IMA-WorldHealth/bhima/issues/6686
Closes https://github.com/IMA-WorldHealth/bhima/issues/6515

**TESTING**
- Use bhima_test (with 'yarn build:db')
- Go to Asset Management > Assets Registry
  - Note that the scan date for MOT1 is in the past
  - Note that the scan date for MOT2 is today
- Go to the Asset Management > Required Inventory Scans page
  - Create a new required inventory scan
    - Insert a title
    - Keep the start day as today
    - Set the completion date to tomorrow
    - Add a description if you wish
    - Click [Submit]
- Go to the Asset Management > Report > Needed Asset Scans
  - Select the Required Inventory Scan
  - Try the three options in the "Inventory Scans Status" field and review the resulting reports
    - Notice the number of assets and scans in the upper right title block
    - In the "All covered inventory articles" -> both MOT1 and MOT1 should be shown
    - In the "Inventory Scanned" => MOT2 should be shown WITH a scan date
    - In the "Unscanned Inventory Articles" report => MOT1 should be shown WITHOUT a scan date (since there has not been a scan in the defined inventory scan period).
